### PR TITLE
lightning: fix panic when table name in source file and target cluster is different (#31808)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #31808 to release-5.2
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In tidb repo:
git pr https://github.com/pingcap/tidb/pull/32366
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/tidb.git pr/32366:release-5.2-167228a2669b
```

---

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/31771

Problem Summary:

Lightning will panic when table name in source file and target cluste…

### What is changed and how it works?

 Fix panic when table name in source file and target cluster is different.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: br/pkg/lightning/restore/tidb_test.go [hunk 1]
```diff
@@ -368,13 +373,23 @@ func (s *tidbSuite) TestLoadSchemaInfo(c *C) {
...
 		},
 	})
 
 	tableCntAfter := metric.ReadCounter(metric.TableCounter.WithLabelValues(metric.TableStatePending, metric.TableResultSuccess))
 
+<<<<<<< HEAD
 	c.Assert(tableCntAfter-tableCntBefore, Equals, 2.0)
+=======
+	require.Equal(t, 3.0, tableCntAfter-tableCntBefore)
+>>>>>>> 167228a26... lightning: fix panic when table name in source file and target cluster is different (#31808)
```
Comment (kennytm) [inline br/pkg/lightning/restore/tidb_test.go:392]: merge conflict

Comment (gozssky) [reply br/pkg/lightning/restore/tidb_test.go]: fixed

**Review by glorv — APPROVED** [review]

**Review by gozssky — APPROVED** [review]

**Pull request merged**